README.md

```diff
@@ -219,7 +219,106 @@ jobs:
 
 ## Outputs
 
-The action prints normal stdout info-level log messages to the actions console. However, it saves a verbose log file to `tower_action_*.log` (the `*` is a timestamp). We recommend using [`actions/upload-artifact`](https://github.com/actions/upload-artifact) in your GitHub Actions workflow as shown in the examples above, this will then expose this file as a download through the workflow summary page.
+### Output variables
+
+The action creates the output variable `json` which is a JSON string of metadata created by the Tower API. It looks like this and can be parsed using the built in `fromJSON()` method.
+
+```
+{
+  "workflowId" : "7f061c344df044",
+  "workflowUrl" : "https://tower.nf/orgs/myorg/workspaces/myworkspace/watch/7f061c344df044",
+  "workspaceId" : 123456789,
+  "workspaceRef" : "[myorg / myworkspace]"
+}
+```
+
+In addition, each variable is available as a separate output available under the following IDs:
+
+- `workflowId`
+- `workflowUrl`
+- `workspaceId`
+- `workspaceRef`
+
+From the example above, we can now extend it to use the output variables in downstream steps and jobs. In this example we use the output variables twice:
+
+- We capture the output variables of the _step_ and use it as an output of the _job_. We then use it in a second job.
+- We use the outputs in a following step where we echo them to the console.
+
+```yaml
+on:
+  push:
+    branches: [dev]
+
+jobs:
+  run-tower:
+    runs-on: ubuntu-latest
+    # Capture action outputs as outputs for the job
+    outputs:
+      workflow_id: ${{ steps.run.outputs.workflowId }}
+      workspace_id: ${{ steps.run.outputs.workspaceId }}
+    steps:
+      - uses: seqeralabs/action-tower-launch@v1
+        with:
+          access_token: ${{ secrets.TOWER_ACCESS_TOKEN }}
+      - uses: actions/upload-artifact@v3
+        id: run # Add id attribute to step
+        with:
+          name: Tower debug log file
+          path: tower_action_*.log
+
+      # Echo variables to console
+      - name: Echo variables
+        run: |
+          echo "Worfklow ID: ${{ steps.run.outputs.workflowId }}"
+          echo "Workflow URL: ${{ steps.run.outputs.workflowUrl }}"
+          echo "Workspace ID: ${{ steps.run.outputs.workspaceId }}"
+          echo "Workspace Reference: ${{ steps.run.outputs.workspaceRef }}"
+
+      # Capture JSON and save as artifact
+      - uses: actions/upload-artifact@v3
+        if: success() || failure()
+        with:
+          name: ${{ needs.getdate.outputs.date }}_run_json
+          path: tower_action_*.json
```

Comment:
I think that we can do this? Or yeah just `tower_action_*`..

```suggestion
          path: |
              tower_action_*.log
              tower_action_*.json
```

Reply:
Yup, but they're both being uploaded as assets for an end-user to download - so user facing it doesn't matter whether that's done as two steps or one, right? Or is there a difference?

Reply:
Yup, but that'd still work with a single upload step, right? Just make it `cat run_details/*json` instead of `cat run_details/*`

I'll prep a PR and we can discuss there.

Reply:
👉🏻  https://github.com/seqeralabs/action-tower-launch/pull/6